### ISSUE TITLE
Cannot create link from Craft entry

### ISSUE THREAD

**Issue by mdunbavan** [jacobgraf/pdfshift#1]
Hi,

I am running the latest plugin release and I am on the latest Craft 3 release.

When I try and get a link to the entry/product I am on I get an error:

```
{{ craft.pdfShift.link({
        source: product.url,
        filename: 'product.pdf',
        format: 'a4',
        sandbox: true
    }) }}
```

The error on the front-end is below:

`Undefined property: stdClass::$url`

Is there a way to fix this or is this something I am doing wrong?

Thanks,
Mark

**Comment by jacobgraf** [comment]
Is the source url you’re using on an internal/development server? PDFShift needs to be able access this source url publicly in order to generate a PDF from it.

When I’m looking to test the PDF on my sites, I either push to a staging server and test it there, or hardcode my source URL to be a link on a staging/live server.

**Comment by mdunbavan** [comment]
@jacobgraf It is on a Craft CMS site and what I am trying to do is enable the user to download the pdf of that page for a product that they are on. The url is on a local version of the site at the moment.

**Comment by jacobgraf** [comment]
Yeah, that's what I was referring to.

If you change your source variable from `product.url` to `'https://www.google.com/'` you'll see that the plugin in working, as intended.

The issue you are seeing is that since PDFShift is an API to a public utility, all the PDF generation stuff happens on the PDFShift servers. If the PDFShift servers can't access the URL you provided to it (your local development machine) it can't generate the PDF. Does that help understand why the error is showing up?

**Comment by mdunbavan** [comment]
Okay so to be clear this plugin doesn't allow you to test it locally in anyway before going live on any of the templates that you have? To test it you need to push it to production for it to work and test it on production?

**Comment by jacobgraf** [comment]
You can certainly test it locally, but you need to either...

1. Hard-code a publicly-accessible URL (e.g. Staging) to test
2. Make your local dev environment accessible from a public URL using something like ngrok or similar (https://ngrok.com/)

Unfortunately, there is no way around this since PDFShift is a 3rd-party, public API that needs visibility to your URL in order to perform its magic.

**Comment by jacobgraf** [comment]
This has been fixed in v1.1.0. Give it an update and let me know how it works. Thanks!
